src/spec/controllers/notifications_controller_spec.rb

```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+
+RSpec.describe NotificationsController do
+  let(:function) { create :function }
+  let(:current_user) { function.user }
+  let(:user) { create :user }
+
+  before { session[:user] = current_user.id }
+
+  describe 'GET #index' do
+    it 'returns unread notifications' do
+      get :index
+      expect(response).to be_successful
+    end
+  end
+
+  describe 'GET #show' do
+    before do
+      old_controller = @controller
+      session[:user] = user.id
+      @controller = LikesController.new
+      post :create, params: { function_id: function, user_id: current_user }
+      session[:user] = current_user.id
+      @controller = old_controller
```

Comment:
you don't need to call the likesController just to create a like and a notification, you can use the FactoryBot factories directly to create the records you need

Reply:
FactoryBot creates the record directly but I need it to be created through the controller because the LikesController trigger the `notify_user` function which creates the notification record